src/librustdoc/clean/utils.rs

```diff
@@ -360,8 +361,8 @@ crate fn resolve_type(cx: &mut DocContext<'_>, path: Path) -> Type {
         Res::SelfTy(..) if path.segments.len() == 1 => Generic(kw::SelfUpper),
         Res::Def(DefKind::TyParam, _) if path.segments.len() == 1 => Generic(path.segments[0].name),
         _ => {
-            let did = register_res(cx, path.res);
-            ResolvedPath { path, did }
+            let _ = register_res(cx, path.res);
```

Comment:
Is this return value used anywhere else? If not I think we can just change `register_res` to remove the return type.

Reply:
Probably we can remove the return value, yeah.

Reply:
It looks like the return value is actually used in a couple places (cleaning imports and resolving intra-doc links). So I'll leave it for now.